features/mbedtls/targets/TARGET_NUVOTON/TARGET_M480/aes/aes_alt.c

```diff
@@ -322,108 +244,138 @@ int mbedtls_aes_crypt_ecb( mbedtls_aes_context *ctx,
  * AES-CBC buffer encryption/decryption
  */
 int mbedtls_aes_crypt_cbc( mbedtls_aes_context *ctx,
-                    int mode,
-                    size_t len,
-                    unsigned char iv[16],
-                    const unsigned char *input,
-                    unsigned char *output )
+                           int mode,
+                           size_t len,
+                           unsigned char iv[16],
+                           const unsigned char *input,
+                           unsigned char *output )
 {
-    unsigned char temp[16];
     int length = len;
-	  int blockChainLen;
-		mbedtls_trace("=== %s [0x%x]\r\n", __FUNCTION__,length);
+    int blockChainLen;
+
     if( length % 16 )
         return( MBEDTLS_ERR_AES_INVALID_INPUT_LENGTH );
 
-    if( (((uint32_t)input) & 0x03) || (((uint32_t)output) & 0x03) )
-		{
-			  blockChainLen = (( length > MAX_DMA_CHAIN_SIZE ) ?  MAX_DMA_CHAIN_SIZE : length );
-    } else {
-			  blockChainLen = length;
-    }			
-		
-    while( length > 0 )
-    {
-			  ctx->opMode = AES_MODE_CBC;
-			  swapInitVector(iv); // iv SWAP
-			  ctx->iv = (uint32_t *)iv;
-
-    		if( mode == MBEDTLS_AES_ENCRYPT )
-    		{					
-	            ctx->encDec = 1;
-	            __nvt_aes_crypt(ctx, input, output, blockChainLen);
-//					    if( blockChainLen == length ) break;		// finish last block chain but still need to prepare next iv for mbedtls_aes_self_test()
-							memcpy( iv, output+blockChainLen-16, 16 );
-				}else{
-					    memcpy( temp, input+blockChainLen-16, 16 );
-		          ctx->encDec = 0;
-	            __nvt_aes_crypt(ctx, input, output, blockChainLen);
-//					    if( blockChainLen == length ) break;		// finish last block chain but still need to prepare next iv for mbedtls_aes_self_test()
-					    memcpy( iv, temp, 16 );
-         }	
-         length -= blockChainLen;
-         input  += blockChainLen;
-         output += blockChainLen;
-			   if(length < MAX_DMA_CHAIN_SIZE ) blockChainLen = length;		// For last remainder block chain				
-	
+
+    ctx->opMode = AES_MODE_CBC;
+    /* Fetch IV byte data in big-endian */
+    ctx->iv[0] = nu_get32_be(iv);
+    ctx->iv[1] = nu_get32_be(iv + 4);
+    ctx->iv[2] = nu_get32_be(iv + 8);
+    ctx->iv[3] = nu_get32_be(iv + 12);
+        
+    if( mode == MBEDTLS_AES_ENCRYPT ) {
+        ctx->encDec = 1;
+    }
+    else {
+        ctx->encDec = 0;
+    }
+    
+    while( length > 0 ) {
+        blockChainLen = (length > MAX_DMA_CHAIN_SIZE) ? MAX_DMA_CHAIN_SIZE : length;
+
+        if( mode == MBEDTLS_AES_ENCRYPT ) {
+            __nvt_aes_crypt(ctx, input, output, blockChainLen);
```

Comment:
This branch is superfluous.

Reply:
@hanno-arm I fixed it in https://github.com/ARMmbed/mbed-os/pull/4925/commits/25fa9bb37c7cf80b0f30f0ff33170fee187227bc.